.github/workflows/build-package.yml

```diff
@@ -0,0 +1,79 @@
+name: Build Package
+
+env:
+  CURRENT_BRANCH: ${GITHUB_REF##*/}
+  GH_TOKEN: "${{ secrets.GH_TOKEN }}"
+  PACKAGING_SECRET_KEY: "${{ secrets.PACKAGING_SECRET_KEY }}"
+  PACKAGE_CLOUD_REPO_NAME: "sample"
+  PACKAGE_ENCRYPTION_KEY: "${{ secrets.PACKAGE_ENCRYPTION_KEY }}"
+  PACKAGING_PASSPHRASE: "${{ secrets.PACKAGING_PASSPHRASE }}"
+on:
+  push:
+    branches: "**"
+
+  workflow_dispatch:
+
+jobs:
+  build_package:
+    name: Build package
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        TARGET_PLATFORM:
+          - centos,8
+          - centos,7
+          - debian,buster
+          - debian,stretch
+          - oraclelinux,8
+          - oraclelinux,7
+          - ubuntu,focal
+          - ubuntu,bionic
+          - ubuntu,xenial
+          - pgxn
+
+    steps:
+      - name: Export target platform
+        run: echo $TARGET_PLATFORM > os-list.csv
+
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install package dependencies
+        run: sudo apt-get update && sudo apt install libcurl4-openssl-dev libssl-dev
+
+      - name: Update and check dockerfiles
+        run: |
+          ./update_dockerfiles
+          git add --intent-to-add dockerfiles
+          git diff --exit-code dockerfiles
+
+      - name: Build images
+        run: |
+          git checkout -- dockerfiles
+          ./update_images
+
+      - name: Clone all-citus branch for test
+        run: git clone -b all-citus --depth=1  https://github.com/citusdata/packaging.git packaging
+
+      - name: Clone tools repo for test
+        run: git clone -b develop_tests --depth=1  https://github.com/citusdata/tools.git packaging/tools
```

Comment:
The branch name will be changed to related tag name after closing the PR below
https://github.com/citusdata/tools/pull/177